File: 0908-middle-of-the-linked-list/0908-middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode first = head;
        ListNode last = head;
        int count = 0;
        while (first != null)
        {
            count++;
            first = first.next;
        }
        count /= 2;
        for (int i = 0; i < count; i++)
        {
            last = last.next;
        }
        return last;
    }
}

File: 0908-middle-of-the-linked-list/0908-middle-of-the-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def middleNode(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        first = head
        last = head
        count = 0
        while first != None:
            count = count + 1
            first = first.next
        count = count / 2
        for i in range(count):
            last = last.next
        return last
        